### PR TITLE
[LPT] [CPU] DepthToSpace transformation

### PR DESCRIPTION
**Pull request by eshoguli**
31553: [CPU plug-in] Implement Reshape-Permute-Reshape pattern to DepthToSpace layer transformation
31559: [LPT] Implement DepthToSpace layer transformation
31554: [CPU plug-in] DepthToSpace node optimizations

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@eshoguli Please keep only LPT related changes here. NG transformation were merged into master.

File: inference-engine/src/low_precision_transformations/src/depth_to_space.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+using namespace InferenceEngine;
+using namespace InferenceEngine::details;
+
+void DepthToSpaceTransformation::transform(TransformationContext& context, CNNLayer& layer) const {
+    if (!canBeTransformed(context, layer)) {
+        return;
+    }
+
+    if ((layer.insData.size() == 0) || layer.insData.size() > 2) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/depth_to_space.cpp:24]: Why you need this check if you already have 
```
    if (parents.size() != 1) {
        return false;
    }
```
on line 45?

File: inference-engine/src/low_precision_transformations/src/depth_to_space.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (C) 2018-2020 Intel Corporation
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/depth_to_space.cpp:1]: 2020

File: inference-engine/src/low_precision_transformations/src/transparent_base_transformation.cpp [hunk 3]
```diff
@@ -29,13 +31,20 @@ void TransparentBaseTransformation::transform(TransformationContext& context, CN
...
+        fillFromDequantizationLayer(*scaleShift, scales, shifts);
 
-        const Blob::Ptr biases_blob = CNNNetworkHelper::getBlob(scaleShift, "biases");
-        auto biases = CNNNetworkHelper::getFloatData(biases_blob);
-        const std::vector<float> shifts = std::vector<float>(biases.get(), biases.get() + biases_blob->size());
+        const size_t inputChannels = CNNNetworkHelper::getInputChannelsCount(layer);
+        const size_t outputChannels = CNNNetworkHelper::getOutputChannelsCount(layer);
+        if (inputChannels != outputChannels) {
+            if (!DequantizationDetails::isPerTensor(scales, shifts)) {
+                THROW_IE_LPT_EXCEPTION(layer) << "layer input and output channels count values are different for by channel quantization";
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/transparent_base_transformation.cpp:42]: Why exception? Max pooling should be handled by transparent transformation and it easily can support per-channel granularity

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp [hunk 4]
```diff
@@ -25,7 +25,7 @@ const std::vector<LayerTransformation::Params> trasformationParamValues = {
 INSTANTIATE_TEST_CASE_P(LPT, ConcatTransformation,
     ::testing::Combine(
         ::testing::ValuesIn(netPrecisions),
-        ::testing::Values(InferenceEngine::SizeVector({ 1, 1024, 16, 16 })),
+        ::testing::Values(InferenceEngine::SizeVector({ 1, 3, 16, 16 })),
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp:28]: Just want to understand why we have this change here?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019 Intel Corporation
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp:1]: 2020

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+// disabled while GPU is not supported DepthToSpace
+INSTANTIATE_TEST_CASE_P(DISABLED_LPT, DepthToSpaceTransformation,
+    ::testing::Combine(
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(InferenceEngine::SizeVector({ 1, 32, 72, 48 })),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU),
+        ::testing::ValuesIn(trasformationParamValues)),
+    DepthToSpaceTransformation::getTestCaseName);
+}  // namespace
+
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp:31]: And again these lines

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/depth_to_space_transformation.cpp [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    const auto reshapeBefore = std::make_shared<ngraph::opset1::Reshape>(fakeQuantize, shapeReshapeBefore, false);
+
+    const auto permutation = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 6 }, { 0, 1, 4, 2, 5, 3 });
+    const auto permute = std::make_shared<ngraph::opset1::Transpose>(reshapeBefore, permutation);
+
+    const auto shapeReshapeAfter = ngraph::opset1::Constant::create(
+        ngraph::element::i64,
+        ngraph::Shape{ 4 },
+        ngraph::Shape{ 1, inputShape[1] / 4ul, inputShape[2] * 2, inputShape[3] * 2 });
+    const auto reshapeAfter = std::make_shared<ngraph::opset1::Reshape>(permute, shapeReshapeAfter, false);
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/depth_to_space_transformation.cpp:71]: Why just not to create DepthToSpace?

Comment (eshoguli) [reply inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/depth_to_space_transformation.cpp:71]: Why not? This is functional tests.

File: inference-engine/src/low_precision_transformations/src/depth_to_space.cpp (hunk 1 above)
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/depth_to_space.cpp:24]: Avoid possible Klockwork issue: check size before usage

File: inference-engine/src/low_precision_transformations/src/transparent_base_transformation.cpp (hunk 3 above)
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/transparent_base_transformation.cpp:42]: Exception happens only if input and output channels values are different

File: inference-engine/src/low_precision_transformations/src/depth_to_space.cpp (hunk 2 above)
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/depth_to_space.cpp:1]: done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp (hunk 5 above)
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp:1]: done

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp (hunk 6 above)
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/depth_to_space_transformation.cpp:31]: done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp (hunk 4 above)
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp:28]: The simplest way to make small independent improvement: test duration time was decreased.

**Pull request merged**